### DISCUSSION TITLE
aws load testing lambda function triggered by iot event

### DISCUSSION THREAD

**Discussion by samuelct01** [artilleryio/artillery#2458] [Artillery]
Hi I'm trying to load test a lambda function I created that is triggered by a JSON message sent via MQTT from aws IOT. Basically I want to simulate iot devices sending mqtt messages to aws IOT core that then passes them to the lambda function and see how the function responds. Is it possilbe to do this with artillery? I am very new to load testing, I've seen some examples of loadtesting lambda functions using http but none with MQTT can anyone point me in the direction of examples or tutorials? thanks

**Comment by hassy** [comment]
hi @samuelct01 👋 you can try this Artillery engine to send MQTT messages to the AWS IoT channel that triggers the Lambda function:

https://github.com/pudkrong/artillery-engine-mqtt
